src/opaque.c

```diff
@@ -1301,11 +1302,11 @@ int opaque_UserAuth(uint8_t _ctx[OPAQUE_SERVER_AUTH_CTX_LEN], const uint8_t auth
 
 // U computes: blinded PW
 // called CreateRegistrationRequest in the ietf cfrg rfc draft
-int opaque_CreateRegistrationRequest(const uint8_t *pw, const uint16_t pwlen, uint8_t _sec[sizeof(Opaque_RegisterUserSec)+pwlen], uint8_t *alpha) {
-  Opaque_RegisterUserSec *sec = (Opaque_RegisterUserSec *) _sec;
-  memcpy(&sec->pw, pw, pwlen);
-  sec->pwlen = pwlen;
-  return blind(pw, pwlen, sec->r, alpha);
+int opaque_CreateRegistrationRequest(const uint8_t *pw, const uint16_t pwlen, uint8_t _ctx[OPAQUE_REGISTER_USER_SEC_LEN+pwlen], uint8_t alpha[crypto_core_ristretto255_BYTES]) {
+  Opaque_RegisterUserSec *ctx = (Opaque_RegisterUserSec *) _ctx;
+  memcpy(&ctx->pw, pw, pwlen);
+  ctx->pwlen = pwlen;
+  return blind(pw, pwlen, ctx->r, alpha);
```

Comment:
i would like to keep the name sec at least in the fn params, as it conveys explicitly that this param contains data that must be kept secret. inside the fn we could rename the param to ctx but i think that is confusing.

Reply:
That makes sense, using `sec` over `ctx`.

I'll revert this change and update `opaque.h` to use `sec` over `ctx` in a separate commit.

In `opaque.h`, the following function pairs use `ctx`:
- `opaque_CreateRegistrationRequest/opaque_FinalizeRequest`
- `opaque_CreateCredentialResponse/opaque_UserAuth`

The following use `sec`:
- `opaque_CreateRegistrationRequest/opaque_FinalizeRequest`
- `opaque_CreateCredentialRequest/opaque_RecoverCredentials`
- `opaque_CreateRegistrationResponse/opaque_StoreUserRecord`

Should we also change the `ctx` of `opaque_CreateCredentialResponse/opaque_UserAuth` to `sec` as well? From what I can tell, this context stays on the server, but the header docs do not mention protecting or sanitizing it.

Reply:
`opaque_FinaleizeRequest` has a `Opaque_RegisterUserSec` as type for the ctx value, which is indeed sensitive, since it contains the password and the blinding factor. so that indeed to be marked as sec.
the ctx in `opaque_UserAuth` has a `Opaque_ServerAuthCTX` type, which has the session authentication key, which is used only once. and only for verification on the server side, however if this key leaks, an attacker can only use it for authentication if they are quicker than the user (which is not difficult) and only if the underlying protocol does not use the shared secret for further, so the case is weak but it still makes sense to also call this param sec instead.

Reply:
I renamed all `ctx` function parameters to `sec`.